File: Source/Ivxr.PlugIndependentLib/WorldModel/Block.cs
using System.Collections.Generic;

namespace Iv4xr.PluginLib.WorldModel
{
    public class Block : Entity
    {
        //public float GridSize;  // NOTE: pull up to Grid if we add grids.
        public float MaxIntegrity;
        public float BuildIntegrity;
        public float Integrity;

        public DefinitionId DefinitionId;

        public PlainVec3D MinPosition;
        public PlainVec3D MaxPosition;
        public PlainVec3I Size;
        public bool Functional;
        public bool Working;
        public List<UseObject> UseObjects;
    }
}

File: JvmClient/src/commonMain/kotlin/spaceEngineers/model/Block.kt
package spaceEngineers.model


interface Block : Entity {
    val maxIntegrity: Float
    val buildIntegrity: Float
    val integrity: Float
    val minPosition: Vec3F
    val maxPosition: Vec3F
    val size: Vec3F
    val useObjects: List<UseObject>
    val functional: Boolean
    val working: Boolean
    val definitionId: DefinitionId
}
